tests/test_payments.py

```diff
@@ -0,0 +1,131 @@
+        )
+        created_records[stream][desc] = card_on_file_payment
+
+        # Create a payment using a gift card
+        desc = "gift_card"
+        gift_card_payment = self.ensure_dict_object(
+            self.client._create_payment(source_key=desc)
+        )
+        created_records[stream][desc] = gift_card_payment
+
+        # Create a payment that will autocomplete
+        desc = "autocomplete"
+        autocomplete_payment = self.ensure_dict_object(
+            self.client._create_payment(autocomplete=True, source_key='card')
+        )
+        created_records[stream][desc] = autocomplete_payment
+
+        # Track all fields from created records
+        fields = set()
+        for desc, record in created_records[stream].items():
+            print("\n\nNew fields found: create of a {} {}".format(desc, stream))
+            if set(record.keys()).difference(fields):
+                print("Adding untracked fields to tracked set: {}\n\n".format(set(record.keys()).difference(fields)))
+            fields.update(record.keys())
+            
+        # Update the records above
+        updated_records = {stream: dict()}
+
+        # Update a completed payment by making a refund (payments must have a status of 'COMPLETED' to process a  )
+        created_desc = "autocomplete"
+        updated_desc = "refund"
+        card_refund, refunded_payment = self.client.create_refund(self.START_DATE, created_records[stream][created_desc])
+        refunded_payment = self.ensure_dict_object(refunded_payment)
+        updated_records[stream][updated_desc] = refunded_payment
+
+        # Update a payment by completing it
+        created_desc = "card_on_file"
+        updated_desc = "complete"
+        obj = created_records[stream][created_desc]
+        completed_payment = self.ensure_dict_object(
+            self.client._update_payment(obj.get('id'), obj=obj, action=updated_desc)
+        )
+        updated_records[stream][updated_desc] = completed_payment
+
+        # Update a payment by canceling it
+        created_desc = "gift_card"
+        updated_desc = "cancel"
+        obj = created_records[stream][created_desc]
+        canceled_payment = self.ensure_dict_object(
+            self.client._update_payment(obj.get('id'), obj=obj, action=updated_desc)
+        )
+        updated_records[stream][updated_desc] = canceled_payment
+
+        # Track all fields from the updated records
+        for desc, record in updated_records[stream].items():
+            if set(record.keys()).difference(fields):
+                print("\n\nNew fields found: update on a {} {}".format(desc, stream))
+                print("Adding untracked fields to tracked set: {}\n\n".format(set(record.keys()).difference(fields)))
+            fields.update(record.keys())
+
+        # Verify all the fields found in the created and updated records are accounted for by the schema
+        schema_keys = set(self.expected_schema_keys(stream))
```

Comment:
So in my testing I found multiple times that it actually failed in the assert dict step because some schema values are not in the top-level keys. Not sure if you want to account for going deeper here?

Reply:
I don't think I want to spend the time on this now. But that is good to know, I hadn't considered subfields with this. Harrison pointed out it is redundant to check against the schema files. I mostly like having it in the all fields test because it prints fields that are in the schema but that we don't get data for as a warning. But this can be accounted for by adding a sync and doing the dict equals like you said. I'll do that.

Reply:
You could argue that's covered by the all fields test, up to you. And yeah, don't need to spend time on it now.